test/units/staking/stakeManager/StakeManager.test.js

```diff
@@ -1443,133 +1472,64 @@ contract('StakeManager', async function(accounts) {
-          assertBigNumberEquality(totalStaked, this.prevTotalStaked.add(auctionDelta))
-        })
-      }
-
-      describe('when 0 dynasties has passed', function() {
-        prepareToTest()
-        testAuctionerIsValidator()
-      })
-
-      describe('when 1000 dynasties has passed', function() {
-        prepareToTest()
-        before(async function() {
-          const currentDynasty = await this.stakeManager.dynasty()
-          await this.stakeManager.advanceEpoch(currentDynasty.mul(new BN('1000')))
-        })
-
-        testAuctionerIsValidator()
-      })
-    })
-
-    describe('when last auctioner is not validator', function() {
-      const heimdallFee = web3.utils.toWei('100')
-
-      function prepareToTest() {
-        before(async function() {
+        before(async function () {
           this.validatorId = '1'
           this.newValidatorId = '3'
-          this.bidder = wallets[3].getChecksumAddressString()
-          this.bidderPubKey = wallets[3].getPublicKeyString()
+          this.bidder = wallets[4].getChecksumAddressString()
+          this.bidderPubKey = wallets[4].getPublicKeyString()
           this.heimdallFee = heimdallFee
           this.bidAmount = bidAmount
         })
 
         before(doDeploy())
 
-        before(async function() {
+        before(async function () {
           this.prevValidatorAddr = initialStakers[0].getChecksumAddressString()
           this.prevValidatorOldBalance = await this.stakeToken.balanceOf(this.prevValidatorAddr)
 
           this.validator = await this.stakeManager.validators(this.validatorId)
         })
       }
 
-      describe('when 0 dynasties has passed', function() {
+      describe('when 0 dynasties has passed', function () {
         prepareToTest()
         testConfirmAuctionBidForNewValidator()
       })
 
-      describe('when 1000 dynasties has passed', function() {
+      describe('when 1000 dynasties has passed', function () {
         prepareToTest()
-        before(async function() {
+        before(async function () {
           const currentDynasty = await this.stakeManager.dynasty()
           await this.stakeManager.advanceEpoch(currentDynasty.mul(new BN('1000')))
         })
 
         testConfirmAuctionBidForNewValidator()
       })
+      describe('when validator has more stake then last bid', function () {
+        prepareToTest()
+        before(async function () {
+          let restakeAmount = web3.utils.toWei('10000')
+          await this.stakeToken.mint(this.prevValidatorAddr, restakeAmount)
+          await this.stakeToken.approve(this.stakeManager.address, restakeAmount, {
+            from: this.prevValidatorAddr
+          })
+          // restake
+          await this.stakeManager.restake(this.validatorId, restakeAmount, false, {
+            from: this.prevValidatorAddr
+          })
+          this.validator = await this.stakeManager.validators(this.validatorId)
+        })
+
+        testConfirmAuctionBidForOldValidator()
```

Comment:
is this.bidderBalanceBeforeAuction updated for this.prevValidatorAddr? can't see here

Reply:
it's in before.